src/TokenHandler/AccountManager.mo

```diff
@@ -128,44 +133,33 @@ module {
 
       if (latestDeposit < prevDeposit) freezeCallback("latestDeposit < prevDeposit on notify");
 
-      let depositInfo = depositRegistry.get(p);
-
       // precredit deposit funds
       if (latestDeposit > fee_) {
-        if (depositInfo.dust == 0) {
+        if (not dust.has(p)) {
```

Comment:
Here, we should still use the registry which is more efficient (log n instead of n). I mean with condition `prevDeposit > fee_`.